### ISSUE TITLE
defaultSerializationFormat property has no effect after upgrade to 2.0.0

### ISSUE THREAD

**Issue by FrVaBe** [camunda/camunda-bpm-spring-boot-starter#185]
After an upgrade of an application from Camunda Spring Boot Starter 1.3.0 to 2.0.0 (each with Camunda 7.6.0) I noticed that the setting of the default serialization format via 
`camunda.bpm.defaultSerializationFormat=application/json` 
has no effect anymore. The serialization format is always _application/x-java-serialized-object_.

I noticed that there is a unit test that covers this setting which passes. But in the end it does not work. I tried to find out what the problem is, but I did not manage to. 

I probably can offer some kind of repro if that helps. But maybe the cause is obvious for you; in that case - I guess - the repo is not needed.

**Comment by jangalinski** [comment]
I will have a look at this

**Comment by FrVaBe** [comment]
I created a repro for this bug in my [camunda/camunda-external-task-demo/](https://github.com/FrVaBe/camunda/tree/master/camunda-external-task-demo) repository. Although I created this repository to do some investigantions on external task usage it contains a unit test that checks the serialization format. Use the [CSBS-Issue-185-repro](https://github.com/FrVaBe/camunda/releases/tag/CSBS-Issue-185-repro) tag to check out the version that just basically handles the issue.

The configuration to use the JSON serialization was done in the `application.properties` file.
The code is camunda spring boot starter version 1.3.0 / 2.0.0 compatible.

The test `DefaultSerializationFormatTest.checkDefaultSerializationFormatHandling` will fail with the
Camunda Spring Boot Starter 2.0.0 version on the classpath (current dependency configuration). If 
Camunda Spring Boot Starter 1.3.0 is used instead, the test will pass (change the dependencies in the pom.xml).

**Comment by aschaeffer** [comment]
As a workaround you can add a component in your application context:

```java
@Component
@Order(Ordering.DEFAULT_ORDER + 1)
public class DefaultSerializationFormatConfiguration implements ProcessEnginePlugin {

	/**
	 * The serialization format to use
	 */
	private static final String DEFAULT_SERIALIZATION_FORMAT = Variables.SerializationDataFormats.JSON.getName();

	@Override
	public final void preInit(final ProcessEngineConfigurationImpl processEngineConfiguration) {
		processEngineConfiguration.setDefaultSerializationFormat(DEFAULT_SERIALIZATION_FORMAT);
	}

	@Override
	public final void postInit(final ProcessEngineConfigurationImpl processEngineConfiguration) {
		// Empty implementation
	}

	@Override
	public final void postProcessEngineBuild(final ProcessEngine processEngine) {
		// Empty implementation
	}

}
```

**Comment by jangalinski** [comment]
The work around does not work for me ...

**Comment by jangalinski** [comment]
I investigated this issue further yesterday ... I created an "example-spin" on the https://github.com/camunda/camunda-bpm-spring-boot-starter/tree/issue185_defaultSerializationFormat branch.
Maybe soneone can have a look at it, it fails due to a missing class "JsonProvider" ... I guess we have a version problem, because camunda spin still uses 0.9.1 and the dependency on the spring boot classpath is 2.2.0 ....
Long story short: I fear this is not related to a code change from 1.3 to 2.0 but due to spring boot version changes in the bom.

**Comment by osteinhauer** [comment]
Hi @jangalinski 

In the branch I _only_ see an added ``org.camunda.bpm.spring.boot.starter.CamundaSerializationFormatIT`` but no "example-spin".

I merged the master into and added
```
<dependency>
    <groupId>org.camunda.spin</groupId>
    <artifactId>camunda-spin-dataformat-all</artifactId>
    <scope>test</scope>
</dependency>
```

The test is OK.

Have I missed something?

**Comment by osteinhauer** [comment]
The effect was a bug in configuration, see #244. Until next release the workaround should be valid. Thx @FrVaBe for providing the example.

**Comment by jangalinski** [comment]
Seems I forgot to push example-spin .... still required?

**Comment by osteinhauer** [comment]
maybe a "example-spin" would be a nice example...

**Comment by jangalinski** [comment]
see #245 ... i'll prepare and push outside this fix release.

**Comment by jangalinski** [comment]
I guess, this can be closed then ...
